app/models/language.rb

```diff
@@ -54,10 +54,14 @@ def explode_path_to_hash key, path
   def self.submission_options
     latest = LanguageGroup.where(identifier: %w[c++ c python haskell java ruby j]).pluck(:current_language_id)
     old = Language.where(identifier: %w[c++11 c++14 c99 python2]).pluck(:id)
-    languages = Language.where(:id => latest).order(:identifier) + Language.where(:id => old).order(:identifier)
+    languages = Language.where(:id => latest + old).order(:name)
```

Comment:
I think it was good having the old languages at the bottom (though a separator might help).
Sorting by name rather than identifier makes sense since that is what the user sees, but I think the order will still be the same because the identifiers and names are similar, right?

Reply:
I think it looks cleaner when sorted fully, and it makes it a bit easier to find a particular option – here's a comparison (I also removed C99): 

![image](https://user-images.githubusercontent.com/5953685/72676779-fdf23c80-3af9-11ea-8c4c-1d1a207d179d.png) ![image](https://user-images.githubusercontent.com/5953685/72676774-eb780300-3af9-11ea-8054-8ba15daa1f99.png)

Sorting by name because when/if we add JavaScript V8 I'm thinking of having the name starting with `JavaScript` but the variant name as something like `v8_8`.

Reply:
Most people shouldn't use the old languages. I agree that it's currently hard to find a particular option, but I think a separator would mostly solve that problem without cluttering up the current languages. Python 3.4 is popular and moves from 6th down to 9th in the comparison (and if we wanted to keep C99 it would be even further down).

Reply:
Having the languages in order benefits those who have never looked at the language list before and are trying to find their preferred language - in which case, an alphabetical ordering is best. 

Everyone else is only interested in selecting their preferred language as quickly as possible - in which case, a popularity/relevance ordering is best.

It depends which case we are optimising for.

Reply:
My ideal ordering would be

C++17
Python 3.4
Java 1.6
Haskell 2010
Ruby 2.2
J 8.0.3
-----
Everything else

Reply:
Have you considered visually separating the new and old languages?

Either using `optgroup` tags or [dividers](https://stackoverflow.com/a/4317135/617159).

Reply:
Not sure how that would translate into ruby. I can make it do this... But the divider is selectable (which isn't ideal). There must be a better way

![ 2020-01-19 at 9 55 20 PM](https://user-images.githubusercontent.com/12654833/72677954-69daa200-3b06-11ea-8197-c0a118dde719.png)

Reply:
I think you'll have to change the template that generates the HTML as well?

Reply:
![ 2020-01-19 at 10 45 05 PM](https://user-images.githubusercontent.com/12654833/72678719-6991d500-3b0d-11ea-8080-f451498f7888.png)

Ignoring the ordering, this is what using option groups looks like

Reply:
Modulo that extra colon, that looks quite nice 🙂

What does everyone else think?

Reply:
I like it as well. @puqeko should be adding this change to #54 shortly.

Reply:
I like the optgroups too.